src/storages/inLocalStorage/SplitsCacheInLocal.ts

```diff
@@ -251,19 +245,8 @@ export class SplitsCacheInLocal extends AbstractSplitsCacheSync {
           // * set change number to -1, to fetch splits with -1 `since` value.
           localStorage.setItem(this.keys.buildSplitsTillKey(), '-1');
 
-          // * remove from cache splits that doesn't match with the new filters
-          this.getSplitNames().forEach((splitName) => {
-            if (queryString && (
-              // @TODO consider redefining `groupedFilters` to expose a method like `groupedFilters::filter(splitName): boolean`
-              groupedFilters.byName.indexOf(splitName) > -1 ||
-              groupedFilters.byPrefix.some((prefix: string) => splitName.startsWith(prefix + '__'))
-            )) {
-              // * set `cacheReadyButNeedsToFlush` so that `checkCache` returns true (the storage is ready to be used) and the data is cleared before updating on first successful splits fetch
-              this.cacheReadyButNeedsToFlush = true;
-              return;
-            }
-            this.removeSplit(splitName);
-          });
+          // * Remove all splits from cache
+          this.removeSplits(this.getSplitNames());
```

Comment:
Check this line. I think this should be `this.clear();` to clear all the cache. You will need to update the E2E tests to see if the behaviour is not broken and it passes edge cases.